File: wwwroot/DesktopModules/RildoInfo/ProductDisplay/View.ascx.cs

using DotNetNuke.Entities.Modules;
using DotNetNuke.Web.Client.ClientResourceManagement;
using System;
using System.Text;

namespace RIW.Modules.ProductDisplay
{
    public partial class View : PortalModuleBase
    {

        private const string RiwModulePathScript = "mod_ModulePathScript";

        public void RegisterModuleInfo()
        {
            //Dim modCtrl As New Entities.Modules.ModuleController()

            dynamic scriptblock = new StringBuilder();
            scriptblock.Append("<script>");
            scriptblock.Append(string.Format("var ascxVersion = \"{0}\"; ", System.IO.File.GetLastWriteTime(string.Format("{0}DesktopModules\\{1}\\View.ascx", Request.PhysicalApplicationPath, this.ModuleConfiguration.DesktopModule.FolderName)).ToString("g")));
            scriptblock.Append(string.Format("var jsVersion = \"{0}\"; ", System.IO.File.GetLastWriteTime(string.Format("{0}DesktopModules\\{1}\\view.js", Request.PhysicalApplicationPath, this.ModuleConfiguration.DesktopModule.FolderName)).ToString("g")));
            scriptblock.Append("</script>");

            // register scripts
            if (!Page.ClientScript.IsClientScriptBlockRegistered(RiwModulePathScript))
            {
                Page.ClientScript.RegisterClientScriptBlock(GetType(), RiwModulePathScript, scriptblock.ToString());
            }
        }

        private void Page_Init(object sender, EventArgs e)
        {
            ClientResourceManager.RegisterStyleSheet(Parent.Page, "/desktopmodules/rildoinfo/webapi/content/owl-carousel/owl.carousel.css", 71);
            ClientResourceManager.RegisterStyleSheet(Parent.Page, "/desktopmodules/rildoinfo/webapi/content/owl-carousel/owl.theme.css", 72);
            ClientResourceManager.RegisterStyleSheet(Parent.Page, "/desktopmodules/rildoinfo/webapi/content/owl-carousel/owl.transitions.css", 73);


            ClientResourceManager.RegisterScript(Parent.Page, "/desktopmodules/rildoinfo/webapi/scripts/owl-carousel/owl.carousel.min.js", 21);
            ClientResourceManager.RegisterScript(Parent.Page, string.Format("DesktopModules/{0}/ViewModels/view.js", this.ModuleConfiguration.DesktopModule.FolderName), 99);
            RegisterModuleInfo();
        }

    }
}

File: wwwroot/DesktopModules/RildoInfo/Reports/Views/View.ascx.vb
Imports RIW.Modules.Reports.Components.Common

Namespace Views

    Partial Class View
        Inherits ReportsModuleBase

        Private Const RiwModulePathScript As String = "mod_ModulePathScript"

#Region "Events"

        Private Sub Page_Init(sender As Object, e As EventArgs) Handles Me.Init
            ClientResourceManagement.ClientResourceManager.RegisterScript(Parent.Page, String.Format("DesktopModules/RildoInfo/{0}/viewmodels/view.js", Me.ModuleConfiguration.DesktopModule.FolderName), 99)
        End Sub

        Private Sub Page_Load(sender As Object, e As EventArgs) Handles Me.Load
            RegisterModuleInfo()
        End Sub

        Public Sub RegisterModuleInfo()

            'Dim modCtrl As New Entities.Modules.ModuleController()

            Dim scriptblock = New StringBuilder()
            scriptblock.Append("<script>")
            scriptblock.Append(String.Format("var ascxVersion = ""{0}""; ", System.IO.File.GetLastWriteTime(String.Format("{0}DesktopModules\RildoInfo\{1}\views\View.ascx", Request.PhysicalApplicationPath, Me.ModuleConfiguration.DesktopModule.FolderName)).ToString("g")))
            scriptblock.Append(String.Format("var jsVersion = ""{0}""; ", System.IO.File.GetLastWriteTime(String.Format("{0}DesktopModules\RildoInfo\{1}\viewmodels\view.js", Request.PhysicalApplicationPath, Me.ModuleConfiguration.DesktopModule.FolderName)).ToString("g")))
            scriptblock.Append("</script>")

            ' register scripts
            If Not Page.ClientScript.IsClientScriptBlockRegistered(RiwModulePathScript) Then
                Page.ClientScript.RegisterClientScriptBlock(GetType(Page), RiwModulePathScript, scriptblock.ToString())
            End If
        End Sub

#End Region

    End Class

End Namespace
